pystac/extensions/timestamps.py

```diff
@@ -0,0 +1,161 @@
+            item.stac_extensions.append(Extensions.TIMESTAMPS)
+
+        self.item = item
+
+    @classmethod
+    def from_item(cls, item):
+        return cls(item)
+
+    @classmethod
+    def _object_links(cls):
+        return []
+
+    def apply(self, published=None, expires=None, unpublished=None):
+        """Applies timestamps extension properties to the extended Item.
+
+        Args:
+            published (datetime or None): Date and time the corresponding data
+                (see below) was published the first time.
+            expires (datetime or None): Date and time the corresponding data
+                (see below) expires (is not valid any longer).
+            unpublished (datetime or None): Date and time the corresponding data
+                (see below) was unpublished.
+        """
+        if published is None and expires is None and unpublished is None:
+            raise pystac.STACError("timestamps extension needs at least one property value.")
+
+        self.published = published
+        self.expires = expires
+        self.unpublished = unpublished
+
+    def _timestamp_getter(self, key, asset=None):
+        if asset is not None and key in asset.properties:
+            timestamp_str = asset.properties.get(key)
+        else:
+            timestamp_str = self.item.properties.get(key)
+
+        timestamp = None
+        if timestamp_str is not None:
+            timestamp = str_to_datetime(timestamp_str)
+
+        return timestamp
+
+    def _timestamp_setter(self, timestamp, key, asset=None):
+        if timestamp is None:
+            self.item.properties[key] = timestamp
+        else:
+            timestamp_str = datetime_to_str(timestamp)
+            if asset is not None:
+                asset.properties[key] = timestamp_str
+            else:
+                self.item.properties[key] = timestamp_str
+
+    @property
+    def published(self):
+        return self.get_published()
+
+    @published.setter
+    def published(self, v):
+        self.set_published(v)
+
+    def get_published(self, asset=None):
+        """Get an Item or Asset published datetime
+
+        If an Asset is supplied and the published property exists on the Asset,
+        return the Asset's value. Otherwise return the Item's value.
+
+        Returns:
+            datetime
+        """
+        return self._timestamp_getter('published', asset)
+
+    def set_published(self, published, asset=None):
+        """Set an Item or asset published datetime
+
+        If an Asset is supplied, sets the property on the Asset.
+        Otherwise sets the Item's value.
+        """
+        self._timestamp_setter(published, 'published', asset)
+
+    @property
+    def expires(self):
+        return self.get_expires()
+
+    @expires.setter
+    def expires(self, v):
+        self.set_expires(v)
+
+    def get_expires(self, asset=None):
+        """Get an Item or Asset expires datetime
```

Comment:
Since the [spec for timestamps](https://github.com/radiantearth/stac-spec/tree/v1.0.0-beta.2/extensions/timestamps#item-fields) states the fields have different meanings if the property is on items or on assets, consider adding some additional documentation about that in these methods that get and set the asset properties.